### PR TITLE
Add laravel 8 support

### PR DESCRIPTION
**Pull request by bend-is**
### Changed

- Minimal required PHP version now is `7.3`
- Minimal `symfony/*` version now is `5.1`
- Minimal `illuminate/*` package versions now is `8.0`
- Minimal `laravel/laravel` package versions now is `8.0`
- Minimal `phpunit/phpunit` package versions now is `9.3`
- Minimal `avto-dev/amqp-rabbit-manager` package versions now is `2.3`
- Version of `php` in docker container updated up to `7.3`
- Version of `composer` in docker container updated up to `2.0.12`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote unit tests for my code _(if tests is required for my changes)_
- [x] I have made changes in `CHANGELOG.md` file

### CODE REVIEW

File: src/Worker.php [hunk 1]
```diff
@@ -4,55 +4,75 @@
...
+use Illuminate\Contracts\Events\Dispatcher;
 use Enqueue\AmqpExt\AmqpContext as Context;
 use Enqueue\AmqpExt\AmqpConsumer as Consumer;
-use Symfony\Component\Debug\Exception\FatalThrowableError;
+use Illuminate\Contracts\Debug\ExceptionHandler;
+use Illuminate\Contracts\Queue\Factory as QueueManager;
 
 class Worker extends \Illuminate\Queue\Worker
 {
+    /** @var string */
```
Comment (eldario) [inline src/Worker.php]: - Split by lines
- Change to `string|null`

File: src/Worker.php [hunk 2]
```diff
@@ -4,55 +4,75 @@
...
-     * @param string        $connectionName
-     * @param string        $queue_names
-     * @param WorkerOptions $options
+     * {@inheritdoc}
+     */
+    public function __construct(QueueManager $manager,
+                                Dispatcher $events,
+                                ExceptionHandler $exceptions,
+                                callable $isDownForMaintenance,
+                                string $consumer_tag = '')
```
Comment (eldario) [inline src/Worker.php]: change to
```php
?string $consumer_tag = null
```

File: src/Worker.php [hunk 3]
```diff
@@ -4,55 +4,75 @@
...
             $last_restart      = (int) $this->getTimestampOfLastQueueRestart();
-            $rabbit_connection = $queue->getRabbitConnection();
-            $rabbit_queue      = $queue->getRabbitQueue();
+            $rabbit_connection = $queue_connection->getRabbitConnection();
+            $rabbit_queue      = $queue_connection->getRabbitQueue();
 
             $consumer   = $rabbit_connection->createConsumer($rabbit_queue);
             $subscriber = $rabbit_connection->createSubscriptionConsumer();
 
+            if ($this->consumer_tag) {
```
Comment (eldario) [inline src/Worker.php]:
```php
if ($this->consumer_tag !== null) {
```

File: Dockerfile [hunk 4]
```diff
@@ -8,7 +8,7 @@ ENV \
     COMPOSER_ALLOW_SUPERUSER="1" \
```
Comment (eldario) [inline Dockerfile]: Remove `COMPOSER_ALLOW_SUPERUSER`

> 
> 1.9.1 2019-11-01
> 
> ...
>     Fixed root warnings always present in Docker containers, setting `COMPOSER_ALLOW_SUPERUSER` is not necessary anymore
> ...

File: README.md [hunk 5]
```diff
@@ -169,7 +169,11 @@ You can dispatch your jobs as usual (`dispatch(new Job)` or `dispatch(new Job)->
 
 #### State storing
 
-Using this package you can store any valiables (except resources and callable entities) between job restarts (just use trait `WithJobStateTrait` in your job class). But you should remember - state is available only inside job `handle` method.
+Using this package you can store any variables (except resources and callable entities) between job restarts (just use trait `WithJobStateTrait` in your job class). But you should remember - state is available only inside job `handle` method.
+
+#### Consumer custom tag prefix
+
+The prefix of the consumer queue tag can be specified with an additional argument in the `AvtoDev\AmqpRabbitLaravelQueue\Worker::__construct` method.
```
Comment (eldario) [inline README.md]: Add something more about this feature.

File: src/Worker.php [hunk 6]
```diff
@@ -4,55 +4,75 @@
...
+                                callable $isDownForMaintenance,
+                                string $consumer_tag = '')
+    {
+        parent::__construct($manager, $events, $exceptions, $isDownForMaintenance);
+
+        $this->consumer_tag = $consumer_tag;
+    }
+
+    /**
+     * Listen to the given queue in a loop.
```
Comment (eldario) [inline src/Worker.php]: Delete it, leave only this line
```
@inheritdoc
```

**Review by eldario — CHANGES_REQUESTED** [review]

File: src/Worker.php (hunk 1 above)
Comment (bend-is) [reply src/Worker.php]: Done

File: src/Worker.php (hunk 2 above)
Comment (bend-is) [reply src/Worker.php]: Done

File: src/Worker.php (hunk 3 above)
Comment (bend-is) [reply src/Worker.php]: Done

File: Dockerfile (hunk 4 above)
Comment (bend-is) [reply Dockerfile]: Done

File: README.md (hunk 5 above)
Comment (bend-is) [reply README.md]: Done

File: src/Worker.php (hunk 6 above)
Comment (bend-is) [reply src/Worker.php]: Done

File: src/Worker.php [hunk 7]
```diff
@@ -4,55 +4,73 @@
...
+    /** @var string|null */
+    protected $consumer_tag;
+
     /**
-     * Listen to the given queue in a loop.
-     *
-     * @param string        $connectionName
-     * @param string        $queue_names
-     * @param WorkerOptions $options
+     * @param string $consumer_tag
```
Comment (eldario) [inline src/Worker.php]:
```suggestion
     * @param string|null $consumer_tag
```

File: src/Worker.php [hunk 8]
```diff
@@ -4,55 +4,73 @@
...
+use Illuminate\Contracts\Events\Dispatcher;
 use Enqueue\AmqpExt\AmqpContext as Context;
 use Enqueue\AmqpExt\AmqpConsumer as Consumer;
-use Symfony\Component\Debug\Exception\FatalThrowableError;
+use Illuminate\Contracts\Debug\ExceptionHandler;
+use Illuminate\Contracts\Queue\Factory as QueueManager;
 
 class Worker extends \Illuminate\Queue\Worker
 {
+    /** @var string|null */
```
Comment (eldario) [inline src/Worker.php]:
```suggestion
    /**
     * @var string|null
     */
```

**Review by eldario — APPROVED** [review]

**Comment by eldario** [comment]
Thanx for your PR, @bend-is

**Pull request merged**